catalogs/catalog-fileset-hadoop/src/main/java/com/datastrato/gravitino/catalog/file/HadoopCatalogOperations.java

```diff
@@ -0,0 +1,383 @@
+        new SchemaEntity.Builder()
+            .withName(ident.name())
+            .withId(stringId.id())
+            .withNamespace(ident.namespace())
+            .withComment(comment)
+            .withProperties(addManagedFlagToProperties(properties))
+            .withAuditInfo(
+                new AuditInfo.Builder()
+                    .withCreator(PrincipalUtils.getCurrentPrincipal().getName())
+                    .withCreateTime(Instant.now())
+                    .build())
+            .build();
+    try {
+      store.put(schemaEntity, true /* overwrite */);
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to create schema " + ident.name(), ioe);
+    }
+
+    return new HadoopSchema.Builder()
+        .withName(ident.name())
+        .withComment(comment)
+        .withProperties(schemaEntity.properties())
+        .withAuditInfo(schemaEntity.auditInfo())
+        .build();
+  }
+
+  @Override
+  public Schema loadSchema(NameIdentifier ident) throws NoSuchSchemaException {
+    try {
+      SchemaEntity schemaEntity = store.get(ident, Entity.EntityType.SCHEMA, SchemaEntity.class);
+
+      return new HadoopSchema.Builder()
+          .withName(ident.name())
+          .withComment(schemaEntity.comment())
+          .withProperties(schemaEntity.properties())
+          .withAuditInfo(schemaEntity.auditInfo())
+          .build();
+
+    } catch (NoSuchEntityException exception) {
+      throw new NoSuchSchemaException("Schema " + ident.name() + " does not exist", exception);
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to load schema " + ident.name(), ioe);
+    }
+  }
+
+  @Override
+  public Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
+      throws NoSuchSchemaException {
+    try {
+      if (!store.exists(ident, Entity.EntityType.SCHEMA)) {
+        throw new NoSuchSchemaException("Schema " + ident.name() + " does not exist");
+      }
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to check if schema " + ident.name() + " exists", ioe);
+    }
+
+    try {
+      SchemaEntity entity =
+          store.update(
+              ident,
+              SchemaEntity.class,
+              Entity.EntityType.SCHEMA,
+              schemaEntity -> updateSchemaEntity(ident, schemaEntity, changes));
+
+      return new HadoopSchema.Builder()
+          .withName(ident.name())
+          .withComment(entity.comment())
+          .withProperties(entity.properties())
+          .withAuditInfo(entity.auditInfo())
+          .build();
+
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to update schema " + ident.name(), ioe);
+    } catch (NoSuchEntityException nsee) {
+      throw new NoSuchSchemaException("Schema " + ident.name() + " does not exist", nsee);
+    } catch (AlreadyExistsException aee) {
+      throw new RuntimeException("Schema " + ident.name() + " already exists", aee);
```

Comment:
The `ident.name()` is NOT the existing schema, we should get it from the exception messages.

Reply:
This actually will never happen, this we doesn't support renaming a schema.